### PR TITLE
'inertia remote add' walkthrough

### LINKED ISSUE #53: Remote Add Walkthrough
We currently silently expect certain parameters on `inertia remote add`. When the user doesnt actually give them, we use defaults which are not good.

Change it to this,

```
$ inertia add remote instance-1
Enter location of pem file:
Enter IP:
```

Then try to extract the SSH user from the pem public file (this has the benefit of failing if we cant find the public key - which would be necessary to authenticate over SSH anyways).

### PR DESCRIPTION
**Pull request by bobheadxi**
:hourglass: **Status**: redy redy

:tickets: **Ticket(s)**: Close #53 'inertia remote add' walkthrough

---

## :construction_worker: Changes

- Walkthrough for the 'remote add' command
- Left `port` as a flag... but maybe should make it part of the walkthrough? 🤔 

## :flashlight: Testing Instructions

```
go install
inertia init
inertia remote add gcloud
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,16 @@ $> inertia init
 
 $> inertia remote add glcoud 35.227.171.49 -u root -i /path/to/my/.ssh/id_rsa
 Remote 'glcoud' added.
+$> inertia remote add gcloud
+Enter location of PEM file (leave blank to use '/Users/yourspecialname/.ssh/id_rsa'):
+/path/to/my/.ssh/id_rsa
+Enter IP address of remote:
+35.227.171.49
+Enter user:
+root
```
Comment (chadlagore) [inline README.md]: So the user can in principle be parsed from the `/Users/yourspecialname/.ssh/id_rsa.pub` file. And in reality, we need to do _some_ verification on whether the user in the public key matches this name. That helps us avoid this type of failure:

```bash
inertia gcloud init
[Permission denied: public key]
```

I bet one of these go libraries we already have as a dependency lets you look at the public key user for a private key 🤔

One nice way to fix:

```
Enter user (press Enter to use default user for this PEM <default_user_we_parsed>):
```

**Comment by chadlagore** [comment]
Might be a good idea to at least ask if they're ok with `8081` as the daemon port.

File: README.md (hunk 1 above)
Comment (bobheadxi) [reply README.md]: @chadlagore If we can read the user of the key from the key, should we even allow manual input of a user?

Comment (chadlagore) [reply README.md]: Its true. If they enter the wrong username, we have no way to recover. Ideally, they don't have too. Another way is they could configure their ssh root once, and we always go rummaging around in there for keys.

File: cmd/remote.go [hunk 2]
```diff
@@ -62,22 +62,48 @@ inerta remote status gcloud`,
...
+		_, err = fmt.Scanln(&response)
+		if err != nil {
+			response = defaultSSHLoc
+		}
+		pemLoc := response
+
+		fmt.Printf("Enter IP address of remote: \n")
+		_, err = fmt.Scanln(&response)
+		if err != nil {
+			log.Fatal("That is not a valid IP address - please try again.")
```
Comment (jordanschalm) [inline cmd/remote.go]: I don't think this is an accurate error message. This error means `Scanln` failed, not that the input wasn't a valid IP

File: cmd/remote.go [hunk 3]
```diff
@@ -62,22 +62,48 @@ inerta remote status gcloud`,
...
+		}
+		address := response
+
+		fmt.Printf("Enter user: \n")
```
Comment (jordanschalm) [inline cmd/remote.go]: Use `Println` and remove the `\n`?

File: cmd/remote.go (hunk 2 above)
Comment (bobheadxi) [reply cmd/remote.go]: An error gets thrown if don't input anything as well (hit 'enter' straight away) - plus anything that causes `Scanln` to fail would be an invalid input right? 😅

File: README.md (hunk 1 above)
Comment (bobheadxi) [reply README.md]: Might leave this for another ticket

Comment (chadlagore) [reply README.md]: 👍

**Pull request merged**